test/libsolidity/SolidityNatspecJSON.cpp

```diff
@@ -485,6 +485,141 @@ BOOST_AUTO_TEST_CASE(event)
 	checkNatspec(sourceCode, "ERC20", userDoc, true);
 }
 
+BOOST_AUTO_TEST_CASE(emit_event_from_another_contract)
+{
+	char const* sourceCode = R"(
+		contract X {
+			/// @notice E event
+			/// @dev E event
+			event E();
+		}
+
+		contract C {
+			function g() public {
+				emit X.E();
+			}
+		}
+	)";
+
+	char const* devDoc = R"ABCDEF(
+	{
+		"events":
+		{
+			"E()":
+			{
+				"details": "E event"
+			}
+		},
+		"kind": "dev",
+		"methods": {},
+		"version": 1
+	}
+	)ABCDEF";
+	checkNatspec(sourceCode, "C", devDoc, false);
+
+	char const* userDoc = R"ABCDEF(
+	{
+		"events":
+		{
+			"E()":
+			{
+				"notice": "E event"
+			}
+		},
+		"kind": "user",
+		"methods": {},
+		"version": 1
+	}
+	)ABCDEF";
+	checkNatspec(sourceCode, "C", userDoc, true);
+}
+
+BOOST_AUTO_TEST_CASE(emit_event_from_another_contract_with_same_signature)
+{
+	char const* sourceCode = R"(
+		contract C {
+			/// @notice C.E event
+			/// @dev C.E event
+			event E(uint256 value);
+		}
+
+		contract D {
+			/// @notice D.E event
+			/// @dev D.E event
+			event E(uint256 value);
+
```

Comment:
Please also add a case where only `C` has Natspec on the event to test Natspec inheritance. That's another thing we introduced in #10996 and then did not test properly when we enabled access to foreign events.

In this case, `D` should **not** inherit Natspec from `C`, because `C` is not in a part of its inheritance hierarchy (as specified in #13086).